Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CSSUnresolvedAbsoluteColor.h"
+
+#include "ColorSerialization.h"
+#include "StyleAbsoluteColor.h"
+#include "StyleBuilderState.h"
+
+namespace WebCore {
+
+void serializationForCSS(StringBuilder& builder, const CSSUnresolvedAbsoluteColor& unresolved)
+{
+    serializationForCSS(builder, unresolved.value);
+}
+
+String serializationForCSS(const CSSUnresolvedAbsoluteColor& unresolved)
+{
+    StringBuilder builder;
+    serializationForCSS(builder, unresolved);
+    return builder.toString();
+}
+
+StyleColor createStyleColor(const CSSUnresolvedAbsoluteColor& unresolved, const Document&, RenderStyle&, Style::ForVisitedLink)
+{
+    return StyleColor {
+        StyleAbsoluteColor {
+            unresolved.value
+        }
+    };
+}
+
+Color createColor(const CSSUnresolvedAbsoluteColor& unresolved, const CSSUnresolvedColorResolutionContext&)
+{
+    return unresolved.value;
+}
+
+bool containsColorSchemeDependentColor(const CSSUnresolvedAbsoluteColor&)
+{
+    return false;
+}
+
+bool containsCurrentColor(const CSSUnresolvedAbsoluteColor&)
+{
+    return false;
+}
```

Comment:
Maybe LTO takes care of this, but I think you’d want these to be inlined.

Reply:
I can also make them constexpr in the header.